File: 55. Jump Game/optimised.py
#!/usr/bin/env python

from typing import List

def canJump(nums: List[int]) -> bool:
    goal = len(nums) - 1

    for i in range(len(nums) -1, -1, -1):
        if i + nums[i] >= goal:
            goal = i

        return goal == 0

print(canJump([2,3,1,1,4]))

File: 55. Jump Game/optimised.ts
function canJump(nums: number[]): boolean {
	const n = nums.length;
	let goal = nums.length - 1;
	for (let i = n - 1; i >= 0; i--) {
		if (i + nums[i] >= goal) goal = i;
	}

	return goal === 0;
};

console.log(canJump([2,3,1,1,4]));
